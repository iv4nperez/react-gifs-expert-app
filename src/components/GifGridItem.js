import React from 'react'

export const GifGridItem = ({ id, title, url }) => {

    return (
        <div className="card animate__animated animate__fadeIn">
           <img className="cover" src={ url } alt={ title } />
           <p>{ title }</p>
        </div>
    )
}
