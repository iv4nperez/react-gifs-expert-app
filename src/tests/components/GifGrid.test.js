import React from 'react';
import  "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas para el componente <GifGrid />', () => {
    
    useFetchGifs.mockReturnValue({
        data:[],
        loading: true
    });

    let category = 'One Punch Man';
    let wrapper = shallow( < GifGrid category={ category } /> );

    test('debe de renderizar el componente correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });
   
    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
       
        const gifs = [
            {
                id: 'ABC',
                url: 'https://localhost/cosa/p.jpg',
                title: 'cualquier cosa'
            },
            {
                id: '123',
                url: 'https://localhost/cosa/p.jpg',
                title: 'cualquier cosa'
            }
        ]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        let wrapper = shallow( < GifGrid category={ category } /> );
        
        // expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    });

    
});