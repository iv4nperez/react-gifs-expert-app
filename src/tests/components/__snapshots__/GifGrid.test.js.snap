// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente <GifGrid /> debe de renderizar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch Man
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
